src/promptflow/promptflow/_sdk/_start_trace.py

```diff
@@ -41,6 +48,36 @@ def start_trace():
     print(f"You can view the trace from UI url: {ui_url}")
 
 
+def _start_pfs_in_background(pfs_port) -> None:
+    """Start a pfs process in background."""
+    if is_port_in_use(pfs_port):
+        _logger.debug("PFS is already serving on port %s", pfs_port)
+        return
+    args = [sys.executable, "-m", "promptflow._sdk._service.entry", "start"]
+    # Start a pfs process using detach mode
+    if platform.system() == "Windows":
+        os.spawnv(os.P_DETACH, sys.executable, args)
+    else:
+        os.system(" ".join(["nohup"] + args + ["&"]))
+
+
+def _liveness_probe(port, interval=60):
```

Comment:
maybe we can leverage the heartbeat API to probe?

Reply:
moved

Reply:
updated according to offline sync